mqtt-client-C-paho/CMakeLists.txt

```diff
@@ -5,4 +5,6 @@ include_directories(/usr/local/include)
 link_directories(/usr/local/lib)
 set(CMAKE_C_STANDARD 99)
 add_executable(mqtt_c main.c)
-target_link_libraries(mqtt_c paho-mqtt3c)
\ No newline at end of file
+target_link_libraries(mqtt_c paho-mqtt3c)
```

Comment:
Thinking we should add an example for using async API as well?

Reply:
I think that should go into another ticket and another PR (I'm not familiar with the async API). @qzhuyan do you want to write a ticket for that?

Reply:
I have done some reading of the POHA async API. It seems to be activated when one sets callback functions. Are you thinking about an async example for the file transfer extension or something else?

Reply:
From the different sources, I have a feeling that paho is promoting async APIs. 
Also, that might be easier to write example code that handles errors such as disconnects, resending file segs etc ....